#36 Feature: register driver & operator

rachelakimana opened the pull request:
#### What does this PR do?

- Create register page

#### Description of Task to be completed? 

- Allow an administrator to register a new driver or a new operator.

- Form validation and display appropriate error messages on unsuccessful attempts

- Responsiveness of the

#### How should this be manually tested?

- Clone the repo

- checkout to branch ``ft-register-36``

- Run `npm install` to install modules

- Run `npm  run start` to start the development server

- On the landing page click on the Sign up menu

#### What are the relevant Pivotal Tracker/trello stories?

- https://trello.com/c/Sd4qDC5s/36-register-drivers-operators

#### Questions:

- N/A

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

creating components of the landing: NavBar, LandiPage and FooterBlock components
adding assets folder and create images folder inside them
including the tailwind css using CDN
trying to configure Tailwind locally using packages but currently THEY ARE NOT WORKING

[ WIP #29 ]

Niyonkuru1 committed:
feat(create UI landing): create UI of the landing page

- Uncommenting the CDN import statement in the index.html

[ WIP #29 ]

Niyonkuru1 committed:
trying to install the tailwind

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

WIP #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

PrinceNiyonshuti commented on src/component/__test__/RegisterPage.test.js:
```diff
@@ -0,0 +1,109 @@
+
+  //   await user.type(screen.getByTestId("email-input"), "rachel@blbla.com");
+  //   await user.type(screen.getByTestId("username-input"), "Rachel");
+  //   await user.type(screen.getByTestId("password-input"), "N#@!Pass");
```
remember to remove the commented codes which are no longer in use

PrinceNiyonshuti reviewed (changes requested):
looks good and as additional remember to provide the expected results as it is in frontend and resolve the conflicts that have been occured

tysepa reviewed (approved):
Please work on that collection @PrinceNiyonshuti has been telling you to collect but you working good

thepatrickniyo reviewed (commented):
Resolve the conflicts as well

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

rachelakimana commented:
@PrinceNiyonshuti @PatrickNiyogitare28 
you can review my PR i have done with the requested changes

thepatrickniyo reviewed (changes requested):
I'm learning it and get weback compilation errors.

PrinceNiyonshuti reviewed (approved):
looks good

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36

thepatrickniyo reviewed (changes requested):
Can you slightly fix these issues? The header should be responsive in your component as well

![image](https://user-images.githubusercontent.com/54368821/166293325-7e04705d-a40e-4f35-9f9d-4fe4d4a7d4ac.png)

And this is how the landing page is looking in your branch, It's not the same case on what we have on develop branch

![image](https://user-images.githubusercontent.com/54368821/166293720-19d08f3b-3950-4417-8fbb-e5dc8d03b5c7.png)

rachelakimana commented:
@PatrickNiyogitare28 , you can review this PR again, I have done with the request changes

rachelakimana committed:
ft(Register): Register Drivers & operators

Create register page and it's responsiveness
display error message on unsuccessful attempts

FINISH #36
